staging/src/k8s.io/apiserver/pkg/apis/audit/types.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package audit
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Level defines the amount of information logged during auditing
+type Level string
+
+// Valid audit levels
+const (
+	// LevelNone disables auditing
+	LevelNone Level = "None"
+	// LevelMetadata provides the basic level of auditing.
+	LevelMetadata Level = "Metadata"
+	// LevelRequestObject provides Metadata level of auditing, and additionally
+	// logs the request object (does not apply for non-resource requests).
+	LevelRequestObject Level = "RequestObject"
```

Comment:
Yeah... we discussed this back in the original proposal [1]. The conclusion there was that this was probably an uncommon use case, and the request could be filtered out (once filters are implemented) if really desired. From an implementation perspective, I think this works since auditing the response is strictly more work on the system than auditing the request.

That said, we could rename the `Response` level to `RequestResponse` (to indicate that both are logged), then if there were sufficient demand we could retroactively add `Response` (_only_ the response) without any other api changes. WDYT?

[1] https://github.com/kubernetes/community/pull/145#discussion_r112614337